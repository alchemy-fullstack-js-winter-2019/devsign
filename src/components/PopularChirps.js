import React from 'react';
import Chirp from './Chirp';

export default function PopularChirps() {
  return (
    <main>
      <h2>Popular Chirps</h2>
      <ul>
        <li><Chirp /></li>
        <li><Chirp /></li>
        <li><Chirp /></li>
        <li><Chirp /></li>
        <li><Chirp /></li>
      </ul>
    </main>
  );
}
