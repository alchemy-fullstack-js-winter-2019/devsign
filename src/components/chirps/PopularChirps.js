import React from 'react';
import PropTypes from 'prop-types';
import Chirp from './Chirp';
import { Main } from './PopularChirpsStyles';
import Loading from '../app/Loading';

export default function PopularChirps({ chirps, term, onChange, loading }) {
  const listOfChirps = chirps.map((chirp, i) => {
    return <li key={i}><Chirp text={chirp.text} handle={chirp.user.nickname} profileImg={chirp.user.picture} /></li>;
  });
  return (
    <Main>
      {loading && <Loading />}
      <h2>Popular Chirps</h2>
      <label><input type="text" name="search" placeholder="Search" value={term} onChange={onChange} /></label>
      <ul>
        {listOfChirps}
      </ul>
    </Main>
  );
}

PopularChirps.propTypes = {
  chirps: PropTypes.array,
  term: PropTypes.string,
  onChange: PropTypes.func,
  loading: PropTypes.bool
};
