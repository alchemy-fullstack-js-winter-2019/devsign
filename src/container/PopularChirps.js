import { connect } from 'react-redux';
import { fetchChirps } from '../actions/chirps';
import { withFetch } from '../components/withFetch';
import Chirps from '../components/chirps/Chirps';
import { getChirps, isChirpsLoading } from '../selectors/chirps';

const mapStateToProps = state => ({
  chirps: getChirps(state),
  loading: isChirpsLoading(state)
});

const mapDispatchToProps = dispatch => ({
  fetch() {
    dispatch(fetchChirps());
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withFetch(Chirps));
