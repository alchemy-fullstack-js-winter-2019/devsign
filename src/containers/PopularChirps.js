import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import Chirps from '../components/Chirps';
import { getAllChirps, isChirpsLoading } from '../selectors/chirps';
import { fetchChirps } from '../actions/chirps';
import PropTypes from 'prop-types';

class PopularChirps extends PureComponent {
  static propTypes = {
    fetch: PropTypes.func.isRequired
  };

  componentDidMount() {
    this.props.fetch();
  }

  render() {
    return <Chirps { ...this.props } />;
  }
}

const mapStateToProps = state => ({
  chirps: getAllChirps(state),
  loading: isChirpsLoading(state)
});

const mapDispatchToProps = dispatch => ({
  fetch() {
    dispatch(fetchChirps());
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PopularChirps);
