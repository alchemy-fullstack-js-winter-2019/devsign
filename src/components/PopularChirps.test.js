import React from 'react';
import renderer from 'react-test-renderer';
import PopularChirps from './PopularChirps';

describe('PopularChirps', () => {
  it('matches a snapshot', () => {
    const tree = renderer.create(
      <PopularChirps />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
