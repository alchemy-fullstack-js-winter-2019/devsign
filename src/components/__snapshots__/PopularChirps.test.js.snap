// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularChirps matches a snapshot 1`] = `
<main>
  <h2>
    Popular Chirps
  </h2>
  <ul>
    <li>
      <section>
        <h4>
          UserName
        </h4>
        <p>
          @userhandle
        </p>
        <p>
          CHIRP CONTENT
        </p>
      </section>
    </li>
    <li>
      <section>
        <h4>
          UserName
        </h4>
        <p>
          @userhandle
        </p>
        <p>
          CHIRP CONTENT
        </p>
      </section>
    </li>
    <li>
      <section>
        <h4>
          UserName
        </h4>
        <p>
          @userhandle
        </p>
        <p>
          CHIRP CONTENT
        </p>
      </section>
    </li>
    <li>
      <section>
        <h4>
          UserName
        </h4>
        <p>
          @userhandle
        </p>
        <p>
          CHIRP CONTENT
        </p>
      </section>
    </li>
    <li>
      <section>
        <h4>
          UserName
        </h4>
        <p>
          @userhandle
        </p>
        <p>
          CHIRP CONTENT
        </p>
      </section>
    </li>
  </ul>
</main>
`;
